src/Model/Model.ts

```diff
@@ -489,16 +489,31 @@ export class Model<T extends IModelData> implements IModel<T> {
     const service = getDataService(this.serviceName);
     observable
       .takeUntil(this.getWillDestroyObservable$())
-      .subscribe((value) => {
+      .subscribe((updatedValue => {
         if (!this.relatedModels.hasOwnProperty(fieldName)) {
-          this.relatedModels[fieldName] = value;
-        } else if (this.relatedModels[fieldName] !== value) {
-          service
-            .actions
-            .setRelationship({ id: this.id, fieldName, value })
-            .invoke();
+          this.relatedModels[fieldName] = updatedValue;
+        } else if (this.relatedModels[fieldName] !== updatedValue) {
+          const currentValue: Model<any> | Model<any>[] | any = this.relatedModels[fieldName];
+
+          if (currentValue instanceof Array && currentValue.some(related => !related.isDestroying)) {
+            currentValue.forEach(related =>
+              // destroy model, so that the subscription above is canceled
+              related.markForDestruction(),
+            );
+            service
+              .actions
+              .setRelationship({ id: this.id, fieldName, value: updatedValue })
+              .invoke();
+          } else if (!currentValue.isDestroying) {
```

Comment:
I was testing this on the SIS this morning and ran into an issue with the if statement logic. When you add a item to an empty collection (for example an insurance when there are none) then try to enter a field on the new item you get an error on line 509 below. What ends up happening is `currentValue instanceof Array` evaluates to true, but `currentValue.some(related => !related.isDestroying)` evaluates to false, so it then tries to evaluate `!currentValue.isDestroying` which is true because `isDestroying` doesn't exist on an array.

The end result is we need to handle the case where `currentValue` is an array, but all items in the array have `isDestroying` set to true.

Reply:
Good catch!